client/src/rr/rdata/ds.rs

```diff
@@ -170,14 +173,11 @@ impl DS {
     /// # Return
     ///
     /// true if and only if the DNSKEY is covered by the DS record.
+    #[cfg(any(feature = "openssl", feature = "ring"))]
     pub fn covers(&self, name: &Name, key: &DNSKEY) -> DnsSecResult<bool> {
         key.to_digest(name, self.digest_type())
             .map_err(|e| e.into())
-            .map(|hash| if &hash as &[u8] == self.digest() {
-                return true;
-            } else {
-                return false;
-            })
+            .map(|hash| hash.as_ref() == self.digest())
```

Comment:
ha, I love it when I forget to just return the result of `==`. This is definitely cleaner.